java/src/main/java/com/google/appengine/tools/pipeline/Job.java

```diff
@@ -385,6 +386,15 @@ private void setCurrentRunGuid(String guid) {
     updateSpec.getNonTransactionalGroup().includeSlot(promisedValue.getSlot());
     return promisedValue;
   }
+  
+  public <F> PromisedValue<F> promise(String promiseHandle) {
```

Comment:
My use case is described in issue #13 and you are not far off.

I have no intention of using the promised value across root jobs, so I have no problem with adding that as a restriction.

Updates soon...